File: ingestion-service/app/models/document.py
from pydantic import BaseModel
from typing import Optional, List, Dict
from datetime import datetime
import uuid

class DocumentMetadata(BaseModel):
    document_id: uuid.UUID
    file_name: str
    file_size: int
    file_type: str
    upload_date: datetime
    last_modified_date: datetime
    user_id: uuid.UUID
    tags: Optional[List[str]] = None
    description: Optional[str] = None
    storage_path: str
    version: int = 1
    checksum: str
    acl: Optional[Dict] = None
    thumbnail_path: Optional[str] = None
    expiration_date: Optional[datetime] = None
    category: Optional[str] = None
    division: Optional[str] = None
    business_unit: Optional[str] = None
    brand_id: Optional[uuid.UUID] = None
    document_type: str
    region: Optional[str] = None
    country: Optional[str] = None
    languages: List[str] = []
    alternate_part_numbers: List[str] = []

class BrandMetadata(BaseModel):
    document_id: uuid.UUID
    available_countries: List[str]
    languages: List[str]
    brand_colors: List[str]
    brand_logo_path: str
    campaign_name: Optional[str] = None
    product_line: Optional[str] = None

File: admin-service/src/types/document.ts
import { v4 as uuid } from 'uuid';

export interface DocumentMetadata {
  document_id: string;
  document_title: string;
  file_name: string;
  file_size: number;
  file_type: string;
  upload_date: Date;
  last_modified_date: Date;
  user_id: string;
  tags: string[];
  description: string;
  storage_path: string;
  version: number;
  checksum: string;
  acl?: { [key: string]: any };
  thumbnail_path?: string;
  expiration_date?: Date;
  category?: string;
  division?: string;
  business_unit?: string;
  brand_id?: string;
  document_type: string;
  region?: string;
  country?: string;
  languages: string[];
  alternate_part_numbers?: string[];
}

export interface BrandMetadata {
  document_id: string;
  available_countries: string[];
  languages: string[];
  brand_colors: string[];
  brand_logo_path: string;
  campaign_name?: string;
  product_line?: string;
}
